src/util.c

```diff
@@ -818,40 +818,47 @@ again:		switch (c = *fmt++) {
             LEGAL_ALT(ALT_O);
             continue;
 
-#ifndef TIME_MAX
-#define TIME_MAX	INT64_MAX
+        case 's': {     /* seconds since the epoch */
+#ifdef _WIN32
+            const time_t TIME_MAX = INT32_MAX;
+#else
+            const time_t TIME_MAX = INT64_MAX;
 #endif
-        case 's':	/* seconds since the epoch */
-            {
-                time_t sse = 0;
-                uint64_t rulim = TIME_MAX;
+            time_t sse;
+            time_t d;
```

Comment:
Instead of adding a cast, I declared `d` as `time_t` instead of `unsigned` so the signedness/size of `TIME_MAX - d` is always correct.